### ISSUE TITLE
Build from source?

### ISSUE THREAD

**Issue by fmo-mt** [bupticybee/TexasSolver#197]
Hey I found that the released version is kinda old and many commits were not included.

Can you provide the ways we can build from source on different platforms?

**Comment by bupticybee** [comment]
Yeah, just load the source code to QT opensource and it should be running. Versions of tool are in readme.

**Comment by bupticybee** [comment]
> > Yeah, just load the source code to QT opensource and it should be running. Versions of tool are in readme.
> 
> I built the latest commit on my M1pro macbook and it runs more than 10x faster than the released one. It's a great work and you should think about updating the releases.

10x faster wow, great news. Could you provide the binary?

**Comment by bupticybee** [comment]
> > > Yeah, just load the source code to QT opensource and it should be running. Versions of tool are in readme.
> > 
> > 
> > I built the latest commit on my M1pro macbook and it runs more than 10x faster than the released one. It's a great work and you should think about updating the releases.
> 
> 10x faster wow, great news. Could you provide the binary?

I don't think any optimization I did over the last year has 10x faster effective, could just be hardware update.

**Comment by fmo-mt** [comment]
you can leave a email here, I'll send you later.

**Comment by bupticybee** [comment]
> you can leave a email here, I'll send you later.

Please send to icybee@yeah.net

**Comment by deardaveed** [comment]
> > Yeah, just load the source code to QT opensource and it should be running. Versions of tool are in readme.
> 
> I built the latest commit on my M1pro macbook and it runs more than 10x faster than the released one. It's a great work and you should think about updating the releases.

can someone post a newer released version for those of us that aren't too tech-savvy?

**Comment by nofreewill42** [comment]
I used chatgpt for help (copy pasting this thread to it and the whole "homepage" of the github project with ctrl+a ctrl+c and pasted to chatgpt as well, then asked how to install this thing on linux)
Gave me `sudo apt-get update
sudo apt-get install build-essential qt5-default cmake git`
But this gave error for qt5-default not existing, copy pasted it to chatgpt and it gave me
`sudo apt-get install qtbase5-dev qtchooser qt5-qmake qtbase5-dev-tools`
which worked.


**So on Linux I've done:**

`(base) nofreewill@nofreewill:~/Escritorio/poker/TexasSolver$ sudo apt-get install qtbase5-dev qtchooser qt5-qmake qtbase5-dev-tools`
`$ qmake TexasSolverGui.pro`
`$ make`

Then double click TexasSolverGui
![image](https://github.com/user-attachments/assets/b2eb964e-b74b-4f98-a28a-223ebc238bfa)



Btw, I just used chatgpt to help me

**Comment by nofreewill42** [comment]
> > > Yeah, just load the source code to QT opensource and it should be running. Versions of tool are in readme.
> > 
> > 
> > I built the latest commit on my M1pro macbook and it runs more than 10x faster than the released one. It's a great work and you should think about updating the releases.
> 
> can someone post a newer released version for those of us that aren't too tech-savvy?

I used chatgpt for help (copy pasting this thread to it and the whole "homepage" of the github project with ctrl+a ctrl+c and pasted to chatgpt as well, then asked how to install this thing on linux)
Gave me `sudo apt-get update
sudo apt-get install build-essential qt5-default cmake git`
But this gave error for qt5-default not existing, copy pasted it to chatgpt and it gave me
`sudo apt-get install qtbase5-dev qtchooser qt5-qmake qtbase5-dev-tools`
which worked.


**So on Linux I've done:**

`(base) nofreewill@nofreewill:~/Escritorio/poker/TexasSolver$ sudo apt-get install qtbase5-dev qtchooser qt5-qmake qtbase5-dev-tools`
`$ qmake TexasSolverGui.pro`
`$ make`

Then double click TexasSolverGui
![image](https://github.com/user-attachments/assets/b2eb964e-b74b-4f98-a28a-223ebc238bfa)

**Comment by ShireGrin** [comment]
Thanks for the example @nofreewill42 

Also if you install open source/community version of qt creator (you can download their online installer) 

And just click on 
![image](https://github.com/user-attachments/assets/bd4ab8a7-754a-4c75-97ad-846e4d7b4910)
Open Project

You should then select the TexasSolverGui.pro file in the repo, that will do everything and index and after a while you can click on the run here:

![image](https://github.com/user-attachments/assets/667e34ef-2ded-48f3-a52e-d649d6d629f6)

**Comment by thx1766** [comment]
I just built the Mac OS version and uploaded it here https://github.com/thx1766/TexasSolver/releases/tag/release-4-july-2025

**Comment by johnwee1** [comment]
I was able to build from source using the above instructions, but I am missing the parameters file for different default ranges, is there anywhere that I can download that from and load it in?

**Comment by Takumi-trbl** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by bupticybee** [comment]
> I was able to build from source using the above instructions, but I am missing the parameters file for different default ranges, is there anywhere that I can download that from and load it in?

Check resource folder it's right there.

**Comment by johnwee1** [comment]
it's not actually there, though I realized I can get the files by downloading the zip file from the last release, the parameters are inside there.

**Comment by bupticybee** [comment]
> it's not actually there, though I realized I can get the files by downloading the zip file from the last release, the parameters are inside there.

Oh then I must put them somewhere else. Anyway good work.  btw do you know the I've released the next gen of texassolver? https://bupticybee.github.io/texassolver_gpu_page/

**Comment by johnwee1** [comment]
im aware, but im on ARM mac, haha...

**Comment by bupticybee** [comment]
> im aware, but im on ARM mac, haha...

Poor john lol

**Comment by deardaveed** [comment]
> I just built the Mac OS version and uploaded it here
> 
> https://github.com/thx1766/TexasSolver/releases/tag/release-4-july-2025
> 
> Used homebrew to get the following (not sure what I already had): $: brew install cmake $: brew install qt5 $: brew install qt $: brew install libomp Move the library binaries (these were the locations for me): $: sudo cp /opt/homebrew/opt/libomp/lib/libomp.a /usr/local/lib/libomp.a $: sudo cp /opt/homebrew/opt/libomp/lib/libomp.dylib /usr/local/lib/libomp.dylib
> 
> It couldnt find omp.h in PCfrSolver.h or BestResponse.h EDIT the files: REMOVE: #include <omp.h> ADD #include "/opt/homebrew/opt/libomp/include/omp.h"
> 
> build it $: qmake TexasSolverGui.pro $: make
> 
> Built on Mac OS 26 Beta 2

wow dope man. any way you can his newest version of the GTO solver for Mac systems as well? Apple Silicone particularly?
